data/data/openstack/masters/main.tf

```diff
@@ -48,6 +48,7 @@ resource "openstack_compute_instance_v2" "master_conf" {
   flavor_id = data.openstack_compute_flavor_v2.masters_flavor.id
   image_id = var.root_volume_size == null ? var.base_image_id : null
   security_groups = var.master_sg_ids
+  availability_zone = var.zones[count.index % length(var.zones)]
```

Comment:
Ok, I fixed all 3 of these by making it so that the fetcher will return []string{""} if there are no zones. That way there is always at least 1 element.